src/client.ts

```diff
@@ -51,6 +54,11 @@ export default class Client implements ClientInterface {
     this.statusItems = new StatusItems(this);
     this.registerCommands();
     this.registerAutoRestarts();
+    this.testController = new TestController(
+      context,
+      this.workingFolder,
+      this.ruby
+    );
```

Comment:
I don't think the `TestController` creation should be the responsibility of the `Client` class, which is supposed to be just the LSP client. I think we can move the creation of the test controller to `extension.ts` and inject it to the client as needed.